From dda151c4601c0f5453ffa7b80f4c77f65ae04bcd Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Mon, 2 Nov 2020 13:47:45 +0100
Subject: [PATCH 5/8] Fix libjpeg workaround for conflict with rpcndr.h

Otherwise it won't compile against mingw-w64-libjpeg-turbo. This version of
libjpeg seems to take care of the conflict already as long as jconfig.h is
included before jmorecfg.h.

Change-Id: I011b7735485689c9797cffe2646c8c3d15d0645e
---
 src/plugins/imageformats/jpeg/qjpeghandler.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/plugins/imageformats/jpeg/qjpeghandler.cpp b/src/plugins/imageformats/jpeg/qjpeghandler.cpp
index a17afc0f69..72708145f4 100644
--- a/src/plugins/imageformats/jpeg/qjpeghandler.cpp
+++ b/src/plugins/imageformats/jpeg/qjpeghandler.cpp
@@ -62,6 +62,7 @@
 // including jpeglib.h seems to be a little messy
 extern "C" {
 #define XMD_H           // shut JPEGlib up
+#include <jconfig.h>    // ensure jconfig.h is included before jmorecfg.h
 #include <jpeglib.h>
 #ifdef const
 #  undef const          // remove crazy C hackery in jconfig.h
-- 
2.35.1

